### DISCUSSION TITLE
How to get the current working directory of the spawned shell

### DISCUSSION THREAD

**Discussion by xq-chia** [Microsoft/node-pty#739] [Q&A] [answered]
## Environment details

- OS: Ubuntu
- OS version: Ubuntu 22.04
- node-pty version: v1.0.0

## Issue description
Is there any way I can get the currect working directory of the spawned pty?

**Comment by Tyriar** [comment, marked as answer]
Short answer is you cannot reliably and cross platform unless you invest in something like a [shell integration script](https://github.com/microsoft/vscode/blob/330ab6c2928963dd83a7d90a271d9f2eb8db90d2/src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-bash.sh#L214).
